src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ErrorNotificationHandlerTest.java

```diff
@@ -30,35 +34,36 @@
 
     private static final ObjectMapper MAPPER = new ObjectMapper();
 
+    private static final UUID LOCK_TOKEN = UUID.randomUUID();
+
+    private static final CompletableFuture<Void> COMPLETED_FUTURE = CompletableFuture.completedFuture(null);
+
     @Mock
     private ErrorNotificationService service;
 
+    @Mock
+    private IQueueClient queueClient;
+
     private ErrorNotificationHandler handler;
 
     @Before
     public void setUp() {
-        handler = new ErrorNotificationHandler(service, MAPPER);
+        handler = new ErrorNotificationHandler(service, MAPPER, new MessageAutoCompletor(queueClient));
     }
 
     @Test
     public void should_fail_exceptionally_when_trying_parse_the_message_body() {
         // given
-        IMessage message = getSampleMessage("{}".getBytes());
+        IMessage message = spy(getSampleMessage("{}".getBytes()));
+        given(message.getLockToken()).willReturn(LOCK_TOKEN);
+        given(queueClient.deadLetterAsync(eq(LOCK_TOKEN), anyString(), anyString())).willReturn(COMPLETED_FUTURE);
 
         // when
         CompletableFuture<Void> future = handler.onMessageAsync(message);
+        future.join();
 
         // then
-        assertThat(future.isCompletedExceptionally()).isTrue();
-
-        // and
-        Throwable throwable = catchThrowable(future::join);
-        assertThat(throwable.getCause())
-            .isInstanceOf(InvalidMessageException.class)
-            .hasMessageContaining("Unable to read error message");
-        assertThat(throwable.getCause().getCause())
-            .isInstanceOf(JsonProcessingException.class)
-            .hasMessageContaining("Missing required creator property 'id'");
+        assertThat(future.isCompletedExceptionally()).isFalse();
```

Comment:
Looks like test name needs updating.
was: `assertThat(future.isCompletedExceptionally()).isTrue();`
is `assertThat(future.isCompletedExceptionally()).isFalse();`